src/agni/server/receiver/UserReceiver.java

```diff
@@ -1,17 +1,55 @@
 package agni.server.receiver;
 
+import java.io.UnsupportedEncodingException;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Vector;
+
 public class UserReceiver implements MessageParser {
-    private UserListener[] userListeners;
+    private Vector <UserListener> userListeners = null;
 
     public UserReceiver() {
-
+        userListeners = new Vector<UserListener>();
     }
 
-    private void notifyUserRequest() {
+    private void notifyUserRequest(String ip, byte type, String message) {
+        for ( UserListener  uListener : userListeners ) {
+            uListener.infoRequest(ip, type, message);
+        }
+    }
 
+    public void register(UserListener uListener) {
+        userListeners.add(uListener);
     }
 
-    public void register(UserListener userListener) {
+    /*
+     * parse ByteBuffer type byte
+     * @requires ByteBuffer Message
+     * @promises user request type as a byte
+     */
+    private byte[] parseMessage(ByteBuffer message) {
+        message.flip();
+        int length = message.remaining();
+        byte[] byteArray = new byte[length];
+        message.get(byteArray);
+        byte[] parsedMessage = Arrays.copyOfRange(byteArray,5,length);
+        return parsedMessage;
+    }
 
+    @Override
+    public void receiveMessage(String ip, ByteBuffer message) {
+    	String userMessage = null;
+        if(ip==null || message == null)
+            throw new IllegalArgumentException();
```

Comment:
If statement should be around curly braces. Also throwing NullPointerException is probably more accurate.